server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedDecoder.java

```diff
@@ -1250,7 +1257,7 @@ void queueResponse(CompletionStage<Object> operationResponse) {
       all.dependsOn(operationResponse.whenComplete(this));
       responseSent = all.freeze()
             .exceptionally(CompletableFutures.toNullFunction())
-            .thenRun(this);
+            .thenRunAsync(this, decoder.responseExecutor);
```

Comment:
This can just be `ch.eventLoop()` so we run all the responses directly in the event loop tied to the channel. It will be automatically ordered as it will enqueue each of the responses in the request order. This way we don't need to use the blocking pool at all.